i18n/zh-cn/docusaurus-plugin-content-docs/current/doc/gb28181.md

```diff
@@ -76,6 +76,53 @@ GB的Candidate定义和[WebRTC: Candidate](./webrtc.md#config-candidate)概念
 
 当然，如果网卡配置了客户端可以访问的地址，可以把`CANDIDATE`配置为`*`，让SRS自己发现。
 
+## External Sip
+
+目前SRS内置的SIP服务器只实现了简单的 `Register`、`Invite` 指令，而如果要实现GB/T-28181的全部功能，势必会引入复杂的上层业务逻辑，甚至一些个性化需求。所以，我们开发了一个独立的外置的SIP服务器。而SRS，只需开放几个简单的API接口，这样就保证了其媒体转发服务器的单一属性，同时兼顾了跟第三方SIP信令服务器对接的需求。
+
+仓库地址位于：https://github.com/ossrs/srs-sip
+
+时序图如下所示：
+```mermaid
+sequenceDiagram
+    Player ->> SRS-SIP : 1. Play Request(with id)
+    SRS-SIP ->>  SRS : 2. Publish Request(with ssrc and id)
+    SRS ->> SRS-SIP : 3. Response(with port)
+    SRS-SIP ->>  Device : 4. Invite(with port)
+    Device ->> SRS-SIP : 5. 200 OK
+    SRS-SIP ->> Player : 6. 200 OK(with url)
+    Device -->> SRS : Media Stream
+    Player ->> SRS : 7. Play
+    SRS -->> Player : Media Stream
+    Player ->> SRS-SIP : 8. Stop Request
+    SRS-SIP ->> SRS : 9. Unpublish Request
+    SRS-SIP ->> Device : 10. Bye
+```
+
+1. 通过SRS-SIP提供的API接口`/srs-sip/v1/invite`，Player主动发起播放请求，携带设备的通道ID
+2. SRS-SIP向SRS发起推流请求，携带SSRC和ID，SSRC是设备推流时RTP里的字段
+3. SRS响应推流请求，并返回收流端口。目前SRS仅支持TCP单端口模式，在配置文件`stream_caster.listen`中配置
+4. SRS-SIP通过GB28181协议向设备发起`Invite`请求，携带SRS的收流端口及SSRC
+5. 设备响应成功
+6. SRS-SIP响应成功，携带URL，用于播放
+7. Player通过返回的URL进行拉流播放
+8. Player停止播放
+9. SRS-SIP通知SRS停止收流
+10. SRS-SIP通过设备停止推流
+
+摄像头上面的配置方法同上，仅需将SIP服务器地址从SRS改成SRS-SIP。
+
+首先启动SRS，请确认版本为`6.0.xxx+`。
+
+然后启动SRS-SIP
+```
+make
+./bin/srs-sip -sip-port 5060 -media-addr 127.0.0.1:1985
+```
+> `media-addr`是SRS的API监听地址
+
+现在，可以通过SRS-SIP内置的网页播放器测试 [http://localhost:6666](http://localhost:6666)
```

Comment:
I tried to compile and run `https://github.com/ossrs/srs-sip`.
There are a few problem.
1. To let build success, the `sipgo` dependence need a local replacement, I just removed the local depdendency.
`replace github.com/emiago/sipgo => ../sipgo`
https://github.com/ossrs/srs-sip/blob/9738a962f7866b5001481d9e009adaddbf684808/go.mod#L18

2. local html server `:6666` never started. don't know what's going wrong.

I used `make` to build in macOS env.

Reply:
1. My mistake that uploaded local configurations by accident.
2. The mistake was that port `6666` couldn't be used, so I changed to a different port. Reference: https://blog.csdn.net/xixiyuguang/article/details/107556211